docs/examples/dali_plugin/custom_copy_client.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python
+
+# The MIT License (MIT)
+#
+# Copyright (c) 2020 NVIDIA CORPORATION
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+# the Software, and to permit persons to whom the Software is furnished to do so,
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import argparse, os, sys
+import numpy as np
+from numpy.random import randint
+import tritongrpcclient
+from PIL import Image
+import math
+
+np.random.seed(100019)
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-v', '--verbose', action="store_true", required=False, default=False,
+                        help='Enable verbose output')
+    parser.add_argument('-u', '--url', type=str, required=False, default='localhost:8001',
+                        help='Inference server URL. Default is localhost:8001.')
+    parser.add_argument('--batch_size', type=int, required=False, default=4,
+                        help='Batch size')
+    parser.add_argument('--n_iter', type=int, required=False, default=-1,
+                        help='Number of iterations , with `batch_size` size')
+    parser.add_argument('--model_name', type=str, required=False, default="mydali",
+                        help='Model name')
+    return parser.parse_args()
+
+
+def array_from_list(arrays):
+    """
+    Convert list of ndarrays to single ndarray with ndims+=1
+    """
+    lengths = list(map(lambda x, arr=arrays: arr[x].shape[0], [x for x in range(len(arrays))]))
+    max_len = max(lengths)
+    arrays = list(map(lambda arr, ml=max_len: np.pad(arr, ((0, ml - arr.shape[0]))), arrays))
+    for arr in arrays:
+        assert arr.shape == arrays[0].shape, "Arrays must have the same shape"
+    return np.stack(arrays)
+
+
+def batcher(dataset, max_batch_size, n_iterations=-1):
```

Comment:
That name is positively foreign. Batcher is a machine for preparing a batch of concrete :)
There are mere 1870 google results for `batcher pytorch`.

Reply:
I used the noun, to reflect that this is not an ordinary function, but a generator. Anyhow, this is a fair point.
@banasraf , what's your opinion? Shall we change this name to `generate_batches`? Or `batch_generator`?

Reply:
No strong opinion. Generally I would prefer verbs for function names, but a generator is kind of different case, so I'm ok with both options

Reply:
I'm fine with both generate_ and _generator. `batcher`- well, as I said, sounds like an industrial machine - I've only found one use of "batcher" in NN context and it wasn't in code written by a native English speaker.

Reply:
Went for `batch_generator`